invokeai/frontend/web/src/features/nodes/components/sidePanel/viewMode/WorkflowWarning.tsx

```diff
@@ -0,0 +1,27 @@
+import { Flex, Icon, Tooltip } from '@invoke-ai/ui-library';
+import { PiWarningBold } from 'react-icons/pi';
+import { WorkflowWarningTooltip } from './WorkflowWarningTooltip';
+import { useGetNodesNeedUpdate } from '../../../hooks/useGetNodesNeedUpdate';
+import { useAppSelector } from '../../../../../app/store/storeHooks';
+import { useMemo } from 'react';
+
+export const WorkflowWarning = () => {
+  const nodesNeedUpdate = useGetNodesNeedUpdate();
+  const isTouched = useAppSelector((s) => s.workflow.isTouched);
+
+  const showWarning = useMemo(() => {
+    return nodesNeedUpdate || isTouched;
+  }, [nodesNeedUpdate, isTouched]);
```

Comment:
The dependencies here are already memoized primitive values and `showWarning` is only used once - wrapping them in a `useMemo` doesn't do anything. Can just `if (!nodeNeedsUpdate && !isTouched) { return null }`